File: development/MODULES/codegen/output/output.js
// Generated Hanami Code (JavaScript)

// Garden: SimpleGarden
class Rose {
    constructor() {
      this.secretNumber = 42;
      this.isFriendly = true;
    }

    sayHello() {
        console.log("Hello from Hanami Rose!\n");
        return;
    }

}

function mainGarden() {
    let userName = undefined;
    console.log("What's your name?");
    // Basic input using prompt:
    userName = parseFloat(prompt()); // Reads string, parses to float
    let g = new Rose();
    g.sayHello();
    if ((userName === "Rose")) {
        console.log("You have a lovely name!\n");
    }
    else if ((userName === "Lily")) {
        console.log("Another beautiful flower name!\n");
    }
    else {
        console.log("Nice to meet you," + userName + "!\n");
    }
    return 0;
}

File: development/MODULES/codegen/output/output.py
class Rose:
    def __init__(self):
      self.secretNumber = 42
      self.isFriendly = True

    def sayHello(self):
        print("Hello from Hanami Rose!\n", end='')
        return


def mainGarden():
    userName = None
    print("What's your name?", end='')
    userName = input() # TODO: Add type conversion if needed (e.g., int(), float())
    g = Rose()
    g.sayHello()
    if (userName == "Rose"):
        print("You have a lovely name!\n", end='')
    elif (userName == "Lily"):
        print("Another beautiful flower name!\n", end='')
    else:
        print("Nice to meet you,", userName, "!\n", end='')
    return 0



if __name__ == "__main__":
    mainGarden()
